rust_src/build.rs

```diff
@@ -110,8 +110,19 @@ where
             },
 
             ParseState::LispFn(name) => {
-                if line.starts_with("pub") || line.starts_with("fn") {
-                    if let Some(func) = name.or_else(|| get_function_name(&line)) {
+                let name = name.or_else(|| get_function_name(&line));
+
+                if line.starts_with("fn") {
+                    panic!(format!(
+                        "
+{} is not public in {} module.
+lisp_fn functions are meant to be used from within remacs as well as in lisp code.
+",
+                        name.unwrap_or("<unknown>".to_string()),
```

Comment:
If `name` is still `None` (can that even happen?) the error isn't very helpful. Do we have a line number?

Reply:
At the moment no. I could try and count.

Reply:
It is highly unlikely that the None case happens.